llm-examples/mpt/mpt-7b/MLFlow Logging and inference with MPT-7B model.py

```diff
@@ -0,0 +1,155 @@
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC Define a customized PythonModel to log into MLFlow.
+
+# COMMAND ----------
+
+import pandas as pd
+import numpy as np
+import transformers
+import mlflow
+import torch
+
+class MPT(mlflow.pyfunc.PythonModel):
+    def load_context(self, context):
+        """
+        This method initializes the tokenizer and language model
+        using the specified model repository.
+        """
+        # Initialize tokenizer and language model
+        self.tokenizer = transformers.AutoTokenizer.from_pretrained(
+          context.artifacts['repository'], padding_side="left")
+
+        config = transformers.AutoConfig.from_pretrained(
+            context.artifacts['repository'], 
+            trust_remote_code=True
+        )
+        # support for flast-attn and openai-triton is coming soon
+        # config.attn_config['attn_impl'] = 'triton'
+        
+        self.model = transformers.AutoModelForCausalLM.from_pretrained(
+            context.artifacts['repository'], 
+            config=config,
+            torch_dtype=torch.bfloat16,
+            trust_remote_code=True)
+        self.model.to(device='cuda')
+        
+        self.model.eval()
+
+    def _build_prompt(self, instruction):
+        """
+        This method generates the prompt for the model.
+        """
+        INSTRUCTION_KEY = "### Instruction:"
+        RESPONSE_KEY = "### Response:"
+        INTRO_BLURB = (
+            "Below is an instruction that describes a task. "
+            "Write a response that appropriately completes the request."
+        )
+
+        return f"""{INTRO_BLURB}
+        {INSTRUCTION_KEY}
+        {instruction}
+        {RESPONSE_KEY}
+        """
+
+    def predict(self, context, model_input):
+        """
+        This method generates prediction for the given input.
+        """
+        generated_text = []
+        for index, row in model_input.iterrows():
+          prompt = row["prompt"]
+          temperature = model_input.get("temperature", [1.0])[0]
+          max_new_tokens = model_input.get("max_new_tokens", [100])[0]
+          full_prompt = self._build_prompt(prompt)
+          encoded_input = self.tokenizer.encode(full_prompt, return_tensors="pt").to('cuda')
+          output = self.model.generate(encoded_input, do_sample=True, temperature=temperature, max_new_tokens=max_new_tokens)
+          prompt_length = len(encoded_input[0])
+          generated_text.append(self.tokenizer.batch_decode(output[:,prompt_length:], skip_special_tokens=True))
+        return pd.Series(generated_text)
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC Download the model
+
+# COMMAND ----------
+
+from huggingface_hub import snapshot_download
+
+# If the model has been downloaded in previous cells, this will not repetitively download large model files, but only the remaining files in the repo
+model_location = snapshot_download(repo_id="mosaicml/mpt-7b-instruct", cache_dir="/local_disk0/.cache/huggingface/")
```

Comment:
please specify the `revision` as well here as recommended behavior to customers :)